torch/_dynamo/variables/tensor.py

```diff
@@ -333,6 +333,23 @@ def method_attr__version(self, tx):
             tx, [self], {}
         )
 
+    def call_hasattr(self, tx, name):
+        try:
+            self.var_getattr(tx, name)
+            ret_val = True
+        except (NotImplementedError, AttributeError):
+            ret_val = False
+
+        if self.source:
+            install_guard(
+                AttrSource(self.source, name).make_guard(GuardBuilder.HASATTR)
+            )
+            ret_val = ret_val or name in self.proxy.node.meta.get("tensor_dict", dict())
+
+        return ConstantVariable(
+            ret_val or tx.output.side_effects.has_pending_mutation_of_attr(self, name)
```

Comment:
Wouldn't `tx.output.side_effects.has_pending_mutation_of_attr(self, name)` be already checked by `var_getattr`? If not, should we move it there.

Reply:
Yeah you're right I think I was just being careful, let me see.